spec/providers/auth.spec.ts

```diff
@@ -61,90 +61,69 @@ describe('AuthBuilder', () => {
-        eventType: 'providers/firebase.auth/eventTypes/user.create',
-        resource: 'projects/myUnitTestProject',
-        notSupported: {
-        },
+    before(() => {
+      cloudFunctionCreate = auth.user().onCreate((ev: Event<firebase.auth.UserRecord>) => ev.data);
+      cloudFunctionDelete = auth.user().onDelete((ev: Event<firebase.auth.UserRecord>) => ev.data);
+      event = {
         data: {
-          uid: 'abcde12345',
-          email: 'foo@bar.baz',
-          emailVerified: false,
-          displayName: 'My Display Name',
-          photoURL: 'bar.baz/foo.jpg',
-          disabled: false,
           metadata: {
             createdAt: '2016-12-15T19:37:37.059Z',
             lastSignedInAt: '2017-01-01T00:00:00.000Z',
           },
-          providerData: [{
-            uid: 'g-abcde12345',
-            email: 'foo@gmail.com',
-            displayName: 'My Google Provider Display Name',
-            photoURL: 'googleusercontent.com/foo.jpg',
-            providerId: 'google.com',
-          }],
-        },
-      };
-
-      const expectedData = {
-        uid: 'abcde12345',
-        email: 'foo@bar.baz',
-        emailVerified: false,
-        displayName: 'My Display Name',
-        photoURL: 'bar.baz/foo.jpg',
-        disabled: false,
-        metadata: {
-          // Gotcha's:
-          // - JS Date is, by default, local-time based, not UTC-based.
-          // - JS Date's month is zero-based.
-          createdAt: new Date(Date.UTC(2016, 11, 15, 19, 37, 37, 59)),
-          lastSignedInAt: new Date(Date.UTC(2017, 0, 1)),
         },
-        providerData: [{
-          uid: 'g-abcde12345',
-          email: 'foo@gmail.com',
-          displayName: 'My Google Provider Display Name',
-          photoURL: 'googleusercontent.com/foo.jpg',
-          providerId: 'google.com',
-        }],
       };
+    });
 
+    it('should transform old wire format for UserRecord into v5.0.0 format', () => {
       return Promise.all([
-        expect(cloudFunctionCreate(event)).to.eventually.deep.equal(expectedData),
-        expect(cloudFunctionDelete(event)).to.eventually.deep.equal(expectedData),
+        cloudFunctionCreate(event).then(data => {
+          expect(data.metadata.creationTime).to.equal('2016-12-15T19:37:37.059Z');
+          expect(data.metadata.lastSignInTime).to.equal('2017-01-01T00:00:00.000Z');
+        }),
+        cloudFunctionDelete(event).then(data => {
+          expect(data.metadata.creationTime).to.equal('2016-12-15T19:37:37.059Z');
+          expect(data.metadata.lastSignInTime).to.equal('2017-01-01T00:00:00.000Z');
+        }),
       ]);
     });
 
-    // This isn't expected to happen in production, but if it does we should
-    // handle it gracefully.
-    it('should tolerate missing fields in the payload', () => {
-      const cloudFunction = auth.user().onCreate((ev: Event<firebase.auth.UserRecord>) => ev.data);
+    it('should still retain createdAt and lastSignedIn', () => {
```

Comment:
A comment on what "createdAt" and "lastSignedIn" are and why they should be retained would be useful.